app/src/lib/trampoline/trampoline-environment.ts

```diff
@@ -64,11 +64,19 @@ export const getCredentialUrl = (cred: Map<string, string>) => {
 
   const protocol = cred.get('protocol') ?? ''
   const username = cred.get('username')
-  const user = username ? `${encodeURIComponent(username)}@` : ''
+  const password = cred.get('password')
+  const user = username
+    ? `${encodeURIComponent(username)}${
+        password ? `:${encodeURIComponent(password)}` : ''
+      }@`
+    : ''
   const host = cred.get('host') ?? ''
+  const port = cred.get('port')
```

Comment:
For future reference I don't believe Git will ever send a `port`. It's included in `host` if there's one.

Reply:
What didn't work? Here's the entirety of what Git supports in a credential: https://github.com/git/git/blob/8b6f19ccfc3aefbd0f22f6b7d56ad6a3fc5e4f37/credential.c#L409-L442